Accept characters in statistic_override (and enable statistic_vertical when passing vectors)

LukasWallrich opened the pull request:
I think it would be helpful to allow character vectors in statistic_override, for instance, to allow for an alternative formatting of CIs. This PR implements that and prints a message to the user notifying that they are printed as given (in case vectors are accidentally formatted as characters).

It also fixes an issue that currently prevents the use of statistic_vertical when passing one vector per model.

As the fix stands, it allows some bad layouts to emerge when statistics_override is used to pass several functions, but the change is minimal. Maybe it should explicitly check whether vectors are passed and only then accept length > 1 - let me know if you would prefer that implementation.

vincentarelbundock commented on R/sanity_checks.R:
```diff
@@ -45,12 +45,12 @@ sanity_checks <- function(models,
     checkmate::assert_logical(statistic_vertical, len = 1, null.ok = FALSE)
     if (!statistic_vertical) {
-        if ((length(statistic_override) > 1) | (length(statistic) > 1)) {
+        if ((length(statistic_override)/length(models) > 1) | (length(statistic) > 1)) {
```
space between operators

vincentarelbundock replied:
Why did this line change?

LukasWallrich replied:
Without that change, I would not be able to pass one vector of (say) standard errors per model and show them next to the estimates.

LukasWallrich commented:
Do you mean _consolidating_ the input checks? Because I think your checks for what statistics_override contains work well (in extract_statistic_override.R), they were just omitting the case of a character vector which then failed later (where I added the code) - or what are you thinking of?

Checking whether the statistics_override argument is suitable for statistic_vertical should probably stay in sanity_checks.R? But indeed, it should be expanded to fail when more than one statistic is specified per model - does that make sense to you?

vincentarelbundock commented:
Yeah, I think I misunderstood what you meant. That sounds good to me. 

I just added a new test file here: https://github.com/vincentarelbundock/modelsummary/blob/master/tests/testthat/test-statistic-override.R

If you can update your branch and add a new test for your new type of input, that would be great. I'd just like all input types to be tested to ensure we don't accidentally break something while adding functionality.

Thanks! Your contributions and ideas are really useful!

LukasWallrich commented:
I've now added the test, edited the documentation to indicate that character vectors are allowed and revised the two edits needed in the existing code. Then I tried to squash the commits and definitely messed up when I rewrote history - can you fix that, or let me know what to do?

LukasWallrich commented:
If this has become too messy, I can also create a new fork and copy the edits into that - just let me know.

vincentarelbundock commented:
You know what, maybe the new fork idea is best. That would allow a clean merge. Sorry for the trouble. Git can be a mess sometimes.

LukasWallrich commented:
Transferred to #54
